cmd/jaeger/internal/exporters/storageexporter/exporter_test.go

```diff
@@ -0,0 +1,131 @@
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	"github.com/jaegertracing/jaeger/model"
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+)
+
+type storageHost struct {
+	storageExtension *component.Component
+	t                *testing.T
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	myMap := make(map[component.ID]component.Component)
+	myMap[jaegerstorage.ID] = *host.storageExtension
+	return myMap
+}
+
+func (host storageHost) ReportFatalError(err error) {
+	host.t.Fatal(err)
+}
+
+func (storageHost) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	return nil
+}
+
+func (storageHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	return nil
+}
+
+func TestExporter(t *testing.T) {
+	exporterFactory := NewFactory()
+
+	ctx := context.Background()
+	telemetrySettings := component.TelemetrySettings{
+		Logger:         zap.L(),
+		TracerProvider: nooptrace.NewTracerProvider(),
+	}
+
+	const memstoreName = "memstore"
+	config := &Config{}
+	config.TraceStorage = memstoreName
+
+	err := config.Validate()
+	require.NoError(t, err)
+
+	tracesExporter, err := exporterFactory.CreateTracesExporter(ctx, exporter.CreateSettings{
+		ID:                ID,
+		TelemetrySettings: telemetrySettings,
+		BuildInfo:         component.NewDefaultBuildInfo(),
+	}, config)
+	require.NoError(t, err)
+
+	extensionFactory := jaegerstorage.NewFactory()
+	storageExtension, err := extensionFactory.CreateExtension(
+		ctx,
+		extension.CreateSettings{
+			TelemetrySettings: telemetrySettings,
+		},
+		&jaegerstorage.Config{Memory: map[string]memoryCfg.Configuration{
+			memstoreName: {MaxTraces: 10000},
+		}})
+	require.NoError(t, err)
+	host := storageHost{
+		storageExtension: &storageExtension,
+		t:                t,
+	}
+
+	err = storageExtension.Start(ctx, host)
+	require.NoError(t, err)
+
+	err = tracesExporter.Start(ctx, host)
+	require.NoError(t, err)
+
+	traces := ptrace.NewTraces()
+	rSpans := traces.ResourceSpans().AppendEmpty()
+	sSpans := rSpans.ScopeSpans().AppendEmpty()
+	span := sSpans.Spans().AppendEmpty()
+
+	spanID := pcommon.NewSpanIDEmpty()
+	spanID[5] = 5 // 0000000000050000
+	span.SetSpanID(spanID)
+
+	traceID := pcommon.NewTraceIDEmpty()
+	traceID[15] = 1 // 00000000000000000000000000000001
+	span.SetTraceID(traceID)
+
+	err = tracesExporter.ConsumeTraces(ctx, traces)
+	require.NoError(t, err)
+
+	storageFactory, err := jaegerstorage.GetStorageFactory(memstoreName, host)
+	require.NoError(t, err)
+	spanReader, err := storageFactory.CreateSpanReader()
+	require.NoError(t, err)
+	requiredTraceID := model.NewTraceID(0, 1) // 00000000000000000000000000000001
+	require.NoError(t, err)
+	requiredTrace, err := spanReader.GetTrace(ctx, requiredTraceID)
+	require.NoError(t, err)
+	assert.Equal(t, spanID.String(), requiredTrace.Spans[0].SpanID.String())
+
+	err = tracesExporter.Shutdown(ctx)
```

Comment:
nit: although doesn't make a difference if the test is fully successful, but if any `require` in the test fails it may leak goroutines. The recommended way is to do `defer te.shutdown()` right after it was successfully created, e.g.
```
defer func() {
  require.NoError(t, tracesExporter.Shutdown(ctx))
}
```